example/src/test/java/com/vivy/springdoc/scoped/startertest/StarterApiTests.java

```diff
@@ -45,4 +59,26 @@ public void apiDocIsGenerated() {
         assertThat(apiDocs.getBody()).contains("paths\":{\"/test\":{\"get");
     }
 
+    @Test
+    public void testThatApiDocFileIsCreatedByNameAndLocationSpecifiedInPropertiesFiles() {
+        String apiDocFileName = environment.getProperty("openapi-boot-starter.swagger-ui.api-docs.file.name");
+        String apiDocFilePath = environment.getProperty("openapi-boot-starter.swagger-ui.api-docs.file.path");
+        Path docApiFile = Paths.get(FileSystems.getDefault().getPath("").toAbsolutePath().toString(), apiDocFilePath, apiDocFileName);
+
+        assertThat(docApiFile).exists();
+    }
+
+    @Test
+    public void testThatTheContentOfTheApiDocCreatedFileContainsBaseURlSpecifiedInThePropertiesFiles() throws IOException {
+        String apiDocFileName = environment.getProperty("openapi-boot-starter.swagger-ui.api-docs.file.name");
+        String apiDocFilePath = environment.getProperty("openapi-boot-starter.swagger-ui.api-docs.file.path");
+        String host = environment.getProperty("openapi-boot-starter.swagger-ui.api-docs.url.host");
+        String servicePort = environment.getProperty("openapi-boot-starter.swagger-ui.api-docs.url.port");
+        Path docApiFile = Paths.get(FileSystems.getDefault().getPath("").toAbsolutePath().toString(), apiDocFilePath, apiDocFileName);
+        String servicePortUrlSegment = servicePort.isBlank() ? servicePort : ":" + servicePort;
+        String openApiSpec = restTemplate.getForEntity("/v3/api-docs", String.class).getBody().replace("http://localhost:" + localServerPort, "http://" + host + servicePortUrlSegment);
```

Comment:
That's not the way it should be done - there is a special config param for a documentation bean to define the proper url

Reply:
sorry, Im not sure if i understant your suggestion would it be ok to have a call? Do tu refere to the replace or the the string concatenation? Initially i was thinking to pars it in to json and then replace the specific field and then save it in a file, but i was thinking maybe it is a bit to overkill, as it will be very

Reply:
this is removed from the scope of this ticket, there is already existing ticket that will handle this